src/types/topology/mod.rs

```diff
@@ -0,0 +1,23 @@
+pub mod routes;
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+type NodeId = u64;
+pub type NodeRef = Rc<RefCell<Node>>;
+
+pub struct Topology {
+    nodes: Vec<NodeRef>,
+}
+
+pub struct Node {
+    name: NodeId,
+    node_type: NodeType,
+    neighbors: HashMap<NodeId, NodeRef>, //node ids
+}
+pub enum NodeType {
```

Comment:
Isn't it repetitive to have NodeId as 'name' in Node, and also in NodeType?

Reply:
You're right

Reply:
Other than that seems like a good start 👍